aten/src/ATen/native/AdaptiveMaxPooling2d.cpp

```diff
@@ -56,15 +56,10 @@ TORCH_META_FUNC(adaptive_max_pool2d) (const Tensor& input, IntArrayRef output_si
 
 TORCH_META_FUNC(adaptive_max_pool2d_backward)
 (const Tensor& grad_output, const Tensor& input, const Tensor& indices) {
+
   int64_t ndim = grad_output.ndimension();
   TORCH_CHECK(ndim == 3 || ndim == 4,
     "adaptive_max_pooling2d_backward(): Expected 3D or 4D grad_output, but got: ", grad_output.sizes());
-  for (const auto i : c10::irange(1, ndim)) {
```

Comment:
Why move the check from `META_FUNC`? The idea is that `META_FUNC` will do the common error checking. This way the CPU and CUDA implementation don't have to add these common checks individually.

So if these as structured kernels, I would suggest that the check should be in `META_FUNC` and not the implementation.

Reply:
If you add the check in `META_FUNC`, then I don't think you'd need to touch the cuda files as `META_FUNC` should apply to both CPU and CUDA.